python/dlisio/plumbing/basicobject.py

```diff
@@ -240,10 +240,10 @@ def fingerprint(self):
 
         fingerprint : str
         """
-        return core.fingerprint(self.type,
-                                self.name,
-                                self.origin,
-                                self.copynumber)
+        return core.fingerprint(self.attic.type,
+                                self.attic.name.id,
+                                self.attic.name.origin,
+                                self.attic.name.copynumber)
```

Comment:
Are these changes still relevant?
They are the same, but the attic version seems more internal than without attic

Reply:
It's more robust to read them from c++. Any encoding changes will take place right away, while using the cached version means the object must be reloaded. Yes, I know we will not (at least for now) be able to support that feature for all strings (fdata_index). But we should improve where we can. Ideally I want to kill the cached name and type in these objects too. But that means updating a lot of tests I think, so that goes in the backlog